### PR TITLE
ZA2-Catalina-Alexandra Jora/ JavaScript-Core-1-Coursework-week1

### PR DESCRIPTION
**Pull request by AlexJora**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

-----
[View rendered exercises/B-hello-world/README.md](https://github.com/AlexJora/JavaScript-Core-1-Coursework-Week1/blob/master/exercises/B-hello-world/README.md)
[View rendered exercises/C-variables/README.md](https://github.com/AlexJora/JavaScript-Core-1-Coursework-Week1/blob/master/exercises/C-variables/README.md)

### CODE REVIEW

**Review by TemwaNyirenda — COMMENTED** [review]
Hey, Catalina, well done on the exercises and mandatory.
There's just a few things to note:
- For exercises/F exercise2: I think your name length will be two too long because you used the myName which had the spaces.
- For extra/2-piping: The functions were supposed to be general so: function add(num1, num2), instead of parameters you used numbers when declaring your function
- For mandatory/1: just before the "years" in string "years old" you need a space otherwise there would be no space between the age and that string
- For mandatory/3: your concatenation won't have any spaces in between the words and I'm not sure that the firstWord would be concatenated using this method

**Comment by Mathias02** [comment]
Well done Catalina, you seem to be a champion in creating variables, other instances where strings, where had to, remain that way but you put everything in a variable, other than that well-done dear,

File: exercises/C-variables/exercise.js [hunk 1]
```diff
@@ -1,3 +1,3 @@
 // Start by creating a variable `greeting`
-
-console.log(greeting);
+let greeting = "Isn't good day at all!"
+console.log(greeting);
```
Comment (chandredewet) [inline exercises/C-variables/exercise.js:3]: Hi Alex,

With this exercise I believe the aim was to print it 3 times. From this I understand you should have had 3 console.log greetings?

I hope the day went better after :)

File: exercises/F-strings-methods/exercise2.js [hunk 2]
```diff
@@ -1,3 +1,11 @@
-const name = " Daniel  ";
+const myName = " Catalina-Alexandra  ";
+const trimmed = myName.trim;
+var message1 = "My name is ";
+var message2 = " and my name is ";
+var message3 = " characters long";
+
+var nameLength = myName.length;
+var message = message1 + trimmed + message2 + nameLength + message3;
```
Comment (chandredewet) [inline exercises/F-strings-methods/exercise2.js:8]: I am just suggesting a shorter way, you could have written this, however all the variables may have made it clearer.  for the trimmed one u could have put myName.trim straight into the message statement. Also u could maybe have just used variables for things that may change. But what u did is also not wrong. Here is just another alternative.

var message = "My name is" + myName.trim + " and my name is " + myName.length + "characters long";

File: exercises/L-functions-nested/exercise.js [hunk 3]
```diff
@@ -3,3 +3,18 @@ var mentor2 = "Irina";
...
+    let shouty = upCase(name);
+    let message = "GOOD DAY " + shouty;
+    return message;
+}
+console.log(greeting(mentor1));
+console.log(greeting(mentor2));
+console.log(greeting(mentor3));
+console.log(greeting(mentor4));
+console.log(greeting(mentor5));
+
```
Comment (chandredewet) [inline exercises/L-functions-nested/exercise.js:20]: Great simple way of doing this project.

Just suggesting alternatives, all the console.logs could have been done in the greeting as well but well done.

File: mandatory/2-logic-error.js [hunk 4]
```diff
@@ -1,16 +1,16 @@
 // The syntax for this function is valid but it has an error, find it and fix it.
 
 function trimWord(word) {
-  return wordtrim();
+  return word.trim();
 }
 
 function getStringLength(word) {
-  return "word".length();
+  return word.length();
```
Comment (chandredewet) [inline mandatory/2-logic-error.js:8]: Did you get this test to pass? As with word.length it is a property and therefore no () at the end - just word,length

**Review by chandredewet — COMMENTED** [review]
Hi Catalina,

Great work as always. I have suggested some changes or alternatives. In some cases as noted, I have suggested alternatives, in other cases, am wondering if the test passed. Well done on doing the Extra exercises as well. I know your javascript will improve as a result.!

**Pull request closed without merging**